kornia/augmentation/container/image.py

```diff
@@ -116,17 +119,121 @@ def __init__(
             ), f"Expect a tuple of (int, int). Got {self.random_apply}."
         else:
             self.random_apply = False
+        self._validate_mix_augmentation(*args, validate_args=validate_args)
+        self.has_mix_augmentation = len(self.__get_mix_indices__(iter(args))) > 0
+
+    def update_attribute(
+        self,
+        same_on_batch: Optional[bool] = None,
+        return_transform: Optional[bool] = None,
+        keepdim: Optional[bool] = None,
+    ) -> None:
+        for arg in self.children():
+            if isinstance(arg, _AugmentationBase):
+                if same_on_batch is not None:
+                    arg.same_on_batch = same_on_batch
+                if return_transform is not None:
+                    arg.return_transform = return_transform
+                if keepdim is not None:
+                    arg.keepdim = keepdim
+
+    @property
+    def same_on_batch(self) -> Optional[bool]:
+        return self._same_on_batch
+
+    @same_on_batch.setter
+    def same_on_batch(self, same_on_batch: Optional[bool]) -> None:
+        self._same_on_batch = same_on_batch
+        self.update_attribute(same_on_batch=same_on_batch)
```

Comment:
are the others set `None` then ?

Reply:
Nope. Nothing will happen if None is passes.